domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/repository/impl/RedisArtworkRepository.kt

```diff
@@ -0,0 +1,131 @@
+package ua.marchenko.artauction.artwork.repository.impl
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.lettuce.core.RedisException
+import java.net.SocketException
+import java.time.Duration
+import org.slf4j.LoggerFactory
+import org.springframework.dao.QueryTimeoutException
+import org.springframework.data.redis.RedisConnectionFailureException
+import org.springframework.data.redis.core.ReactiveRedisTemplate
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.kotlin.core.publisher.toMono
+import ua.marchenko.artauction.artwork.model.MongoArtwork
+import ua.marchenko.artauction.artwork.model.projection.ArtworkFull
+import ua.marchenko.artauction.artwork.repository.ArtworkRepository
+import ua.marchenko.core.artwork.enums.ArtworkStatus
+import ua.marchenko.core.artwork.exception.ArtworkNotFoundException
+
+@Repository
+internal class RedisArtworkRepository(
+    private val mongoArtworkRepository: MongoArtworkRepository,
+    private val reactiveRedisTemplate: ReactiveRedisTemplate<String, ByteArray>,
+    private val objectMapper: ObjectMapper,
+) : ArtworkRepository by mongoArtworkRepository {
+
+    override fun save(artwork: MongoArtwork): Mono<MongoArtwork> =
+        mongoArtworkRepository.save(artwork)
+            .flatMap { savedArtwork ->
+                val id = requireNotNull(savedArtwork.id) { "artwork id cannot be null" }.toHexString()
```

Comment:
seems, `requireNotNull` is redundant. `id` here cannot be null

Reply:
I have nullable id in `MongoArtwork`, so I need to check the id for nullability